app-backend/api/src/main/scala/annotation-project/AnnotationProjectTaskRoutes.scala

```diff
@@ -0,0 +1,362 @@
+              .transact(xa)
+              .unsafeToFuture
+          )
+        }
+      }
+    }
+  }
+
+  def deleteTasks(projectId: UUID): Route = authenticate { user =>
+    authorizeScope(
+      ScopedAction(Domain.AnnotationProjects, Action.DeleteTasks, None),
+      user
+    ) {
+      authorizeAuthResultAsync {
+        AnnotationProjectDao
+          .authorized(
+            user,
+            ObjectType.AnnotationProject,
+            projectId,
+            ActionType.Edit
+          )
+          .transact(xa)
+          .unsafeToFuture
+      } {
+        complete {
+          TaskDao
+            .deleteProjectTasks(projectId)
+            .transact(xa)
+            .unsafeToFuture map { _ =>
+            HttpResponse(StatusCodes.NoContent)
+          }
+        }
+      }
+    }
+  }
+
+  def createTaskGrid(projectId: UUID): Route = authenticate { user =>
+    authorizeScope(
+      ScopedAction(Domain.AnnotationProjects, Action.CreateTaskGrid, None),
+      user
+    ) {
+      authorizeAuthResultAsync {
+        AnnotationProjectDao
+          .authorized(
+            user,
+            ObjectType.AnnotationProject,
+            projectId,
+            ActionType.Edit
+          )
+          .transact(xa)
+          .unsafeToFuture
+      } {
+        entity(as[Task.TaskGridFeatureCreate]) { tgf =>
+          complete(
+            StatusCodes.Created,
+            TaskDao
+              .insertTasksByGrid(
+                Task.TaskPropertiesCreate(TaskStatus.Unlabeled, projectId),
+                tgf,
+                user
+              )
+              .transact(xa)
+              .unsafeToFuture
+          )
+        }
+      }
+    }
+  }
+
+  def getTaskUserSummary(projectId: UUID): Route = authenticate { user =>
+    authorizeScope(
+      ScopedAction(Domain.AnnotationProjects, Action.ReadTasks, None),
+      user
+    ) {
+      {
+        authorizeAuthResultAsync {
+          AnnotationProjectDao
+            .authorized(
+              user,
+              ObjectType.AnnotationProject,
+              projectId,
+              ActionType.Annotate
+            )
+            .transact(xa)
+            .unsafeToFuture
+        } {
+          (userTaskActivityParameters) { userTaskActivityParams =>
+            complete {
+              TaskDao
+                .getTaskUserSummary(
+                  projectId,
+                  userTaskActivityParams
+                )
+                .transact(xa)
+                .unsafeToFuture
+            }
+          }
+        }
+      }
+    }
+  }
+
+  def getTask(projectId: UUID, taskId: UUID): Route = authenticate { user =>
+    authorizeScope(
+      ScopedAction(Domain.AnnotationProjects, Action.ReadTasks, None),
+      user
+    ) {
+      {
+        authorizeAuthResultAsync {
+          AnnotationProjectDao
+            .authorized(
+              user,
+              ObjectType.AnnotationProject,
+              projectId,
+              ActionType.Annotate
```

Comment:
Unless it's expressing something other than what I think it's expressing :thinking:

Reply:
@jisantuc Ah, I think I see where this comes from. So for all these ACR level permissions, I just applied the same actions as what we did for project routes, which had to differentiate a `View` and an `Annotate` action for different app users. So basically, only for endpoints that were written specifically for Annotate, we had applied `Annotate` action or above (by above, I mean, `Edit` and `Delete`). Since the `acrs` on `annotation_projects` were direct copies from `projects`, my thought was that having the same allowed actions on endpoints would not cause unexpected returns or `403`s -- this was the reasoning behind these `Annotate` actions being on these routes. 

Now that `AnnotationProject` is a first-class object in terms of permissions, having a `View` action also makes sense since users without annotate scope will not even get to this point. I kept the same action just for consistency, and being in a labeler/validator team gives you `View`, `Annotate`, and `Edit` permissions.